pyspatialopt/models/binary_mclp_distance_matrix.py

```diff
@@ -0,0 +1,128 @@
+        output["totalDemand"] += row["demand"]    
+    logging.getLogger().info("Binary coverage successfully generated.")
+    return output
+
+def binary_mclp_distance_matrix(file_distance_matrix, service_dist, num_facility, list_field_req = ["facility_id", "demand_id", "demand", "distance"], facility_variable_name = "facility", workspace_path = "."):
+    """
+    Solve a binary and point-based MCLP based on a distance matrix
+    :param file_distance_matrix: (string) file name of a distance matrix. CSV format.
+    :param service_dist: (numeric) maximum service distance
+    :param num_facility: (integer) number of facilities to locate
+    :param list_field_req: (list of string) a list of fields in the file_distance_matrix
+    :param facility_variable_name: (string) facility variable name in the coverage object
+    :param workspace_path: (string) the folder path of file_distance_matrix
+    :return: (dictionary) A dictionary storing the coverage result
+    """
+    # read the distance matrix
+    with open(os.path.join(workspace_path, file_distance_matrix)) as csvfile:
+        dict_pairwise_distance = [{k: v for k, v in row.items()}
+        for row in csv.DictReader(csvfile, skipinitialspace=True)]
+
+    # The file should contain the required fields. If not, exit
+    item_pairwise_distance = dict_pairwise_distance[1]
+    
+    for field in list_field_req:
+        if field not in item_pairwise_distance.keys():
+            print("Error: this field {} not found in the distance csv".format(field))
+            sys.exit(0)
+
+    # creat a coverage object. Need to write a new function
+    dict_coverage = generate_binary_coverage_from_dist_matrix(list_dict_facility_demand_distance = dict_pairwise_distance, dl_id_field = "demand_id", fl_id_field = "facility_id", dist_threshold = service_dist, demand_field="demand", distance_field="distance", fl_variable_name=facility_variable_name)        
+    
+    # formulate model
+    mclp = covering.create_mclp_model(dict_coverage, {"total": num_facility})
+
+    # solve
+    mclp.solve(pulp.GLPK())
+
+    # Get the id set of facilities chosen
+    set_facility_id_chosen = set(utilities.get_ids(mclp, facility_variable_name))
+    
+    # Query the demand covered from the dict_coverage
+    total_demand_covered = 0.0
+
+    for demand_id, demand_obj in dict_coverage["demand"].items():
+        # if this demand_id is covered by any facility in ids
+        if not set_facility_id_chosen.isdisjoint(demand_obj["coverage"]["facility"].keys()):
+            total_demand_covered += demand_obj["demand"]
+    
+    result_coverage = {"number_facility": num_facility,
+    "number_facility_chosen": len(set_facility_id_chosen),
+    "set_facility_id_chosen": set_facility_id_chosen,
+    "total_demand": dict_coverage["totalDemand"], 
+    "percent_demand_coverage":(100 * total_demand_covered) / dict_coverage["totalDemand"]
+    }
+
+    return result_coverage
+    
```

Comment:
Delete these extra ilnes here. There should be one newline after your `return` statement per PEP 8.

Reply:
A bit confused here: in pep8 it says "Surround top-level function and class definitions with two blank lines". Looks like two blank lines are needed after the `return` statement?